src/Features/CSharp/Portable/AddBraces/CSharpAddBracesDiagnosticAnalyzer.cs

```diff
@@ -74,9 +88,17 @@ public void AnalyzeNode(SyntaxNodeAnalysisContext context)
                     {
                         return;
                     }
+
                     break;
             }
 
+            if (option.Value == PreferBracesPreference.WhenMultiline
+                && !IsConsideredMultiLine(statement, embeddedStatement)
+                && !RequiresBracesToMatchContext(statement, embeddedStatement))
```

Comment:
probably worth commenting the overall concept here, esp. about the bit about 'matching context'.

Reply:
📝 Hoping the new documentation for each of the two methods called here makes the behavior clear, but awaiting confirmation.

Reply:
Yup.  The docs was *super* helpful. I think it could be improved a *small* amount with some naming tweaks, but overall i both really like the intuitive approach here, and believe the code is doing the right thing.